utils/studyFunctions.js

```diff
@@ -0,0 +1,156 @@
+const currentDate = new Date();
+
+export function generateStudyMode({ wordList, setStudyList, setIsStudyMode }) {
+  const entriesDue = wordList.filter((entry) => {
+    return isDue(entry.study.lastReview, entry.study.stage);
+  });
+
+  setStudyList(entriesDue);
+  setIsStudyMode(true);
+
+  return entriesDue;
+}
+
+function isDue(lastReview, stage) {
+  if (lastReview === "new") {
+    return true;
+  }
+
+  let interval = 0;
+  switch (stage) {
+    case 0:
+    case 1:
+      interval = 1;
+      break;
+    case 2:
+      interval = 2;
+      break;
+    case 3:
+      interval = 7;
+      break;
+    case 4:
+      interval = 14;
+      break;
+    case 5:
+      interval = 30;
+      break;
+    case 6:
+      interval = 120;
+      break;
+    case 7:
+      interval = "done";
+      break;
+  }
+
+  if (interval === "done") {
+    return false;
+  }
+
+  const reviewDate = new Date(lastReview);
+  if (interval >= 0 && interval <= 120) {
+    const dueDate = new Date(
+      reviewDate.setDate(reviewDate.getDate() + interval)
+    );
+
+    if (currentDate >= dueDate) {
+      return true;
+    }
+  }
+  return false;
+}
+
+export function handleRightAnswer({
+  studyList,
+  setIsFront,
+  entry,
+  databaseMutate,
+}) {
+  studyList.shift();
+  setIsFront(true);
+
+  const stageModifier = 1;
+  const wasWrongAnswer = false;
+  handleUpdateEntryInStudyMode({
+    entry,
+    stageModifier,
+    wasWrongAnswer,
+    databaseMutate,
+  });
+}
+
+export function handleWrongAnswer({
+  studyList,
+  setIsFront,
+  entry,
+  databaseMutate,
+}) {
+  studyList.shift();
+  setIsFront(true);
+
+  const stageModifier = -1;
+  const wasWrongAnswer = true;
+  handleUpdateEntryInStudyMode({
+    entry,
+    stageModifier,
+    wasWrongAnswer,
+    databaseMutate,
+  });
+}
+
+async function handleUpdateEntryInStudyMode({
+  entry,
+  stageModifier,
+  wasWrongAnswer,
+  databaseMutate,
+}) {
+  const { _id, study } = entry;
+  const { stage, wrongAnswerCount, rightAnswerCount, streak } = study;
+
+  const newStreak =
+    wasWrongAnswer && streak > 0
+      ? 0
+      : wasWrongAnswer && streak <= 0
+      ? streak - 1
+      : !wasWrongAnswer && streak < 0
+      ? 0
+      : streak + 1;
```

Comment:
No nested ternary world championship record please : D

Write an additional function outside with explicit `if` statements ^^

Reply:
I changed it to functions :)